### PR TITLE
feat: support multiple roots in React and Vue selectors

### LINKED ISSUE #8230: Using React Selectors with multiple React trees
Hello,
Excited to see the new React Selector api! Our site uses two react trees and I wondering if there can be support for multiple trees.

I have my own custom react selector on top of the Selector api and this is what I do today:
image

Thanks!

### PR DESCRIPTION
**Pull request by aslushnikov**
Fixes #8230

### CODE REVIEW

**Review by JoelEinbinder — APPROVED** [review]
cool!

File: src/server/injected/vueSelectorEngine.ts [hunk 1]
```diff
@@ -233,9 +240,8 @@ export const VueEngine: SelectorEngine = {
         if (!checkComponentAttribute(treeNode.props, attr))
           return false;
       }
-
       return true;
-    });
+    })).flat();
```
Comment (dgozman) [inline src/server/injected/vueSelectorEngine.ts:244]: Is this available in all our browsers?

**Review by dgozman — APPROVED** [review]
Let's add a test where I can find something in a particular root, like `#root1 >> react=MyBook` or something like that.

File: src/server/injected/vueSelectorEngine.ts (hunk 1 above)
Comment (aslushnikov) [reply src/server/injected/vueSelectorEngine.ts:244]: yes

**Pull request merged**